include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -559,13 +559,15 @@ class alignment_file_input_format_REMOVEME<format_sam>
         // check for soft clipping at the first two positions
         if (!cigar_vector.empty() && 'S'_cigar_op == cigar_vector[0])
             sc_begin = get<0>(cigar_vector[0]);
-        else if (cigar_vector.size() > 1 && 'S'_cigar_op == cigar_vector[1])
+        else if (cigar_vector.size() > 1 && 'H'_cigar_op == cigar_vector[0] && 'S'_cigar_op == cigar_vector[1])
             sc_begin = get<0>(cigar_vector[1]);
 
         // check for soft clipping at the last two positions
         if (cigar_vector.size() > 1 && 'S'_cigar_op == cigar_vector[cigar_vector.size() - 1])
             sc_end = get<0>(cigar_vector[cigar_vector.size() - 1]);
-        else if (cigar_vector.size() > 2 && 'S'_cigar_op == cigar_vector[cigar_vector.size() - 2])
+        else if (cigar_vector.size() > 2 &&
+                 'H'_cigar_op == cigar_vector[cigar_vector.size() - 1] &&
+                 'S'_cigar_op == cigar_vector[cigar_vector.size() - 2])
```

Comment:
not sure but can you check that if the condition block spans more than one line whether we have to wrap the expression in `{}` or not? I had the impression that when ever a if/for etc. expression spans multiple lines to wrap it in braces.

Reply:
I don't find this specific. What would you like?

Reply:
@rrahn ping

Reply:
```cpp
if (expression too
     long)
{
  // make braces
}
```

Reply:
I made the expression short by caching the vector size. take a look. I think this is even more readable.